core/src/common.cpp

```diff
@@ -376,8 +376,8 @@ namespace irods
 
 		irods::at_scope_exit clear_options{[&input] { clearKeyVal(&input.options); }};
 
-		std::strncpy(input.username, _username.c_str(), sizeof(SwitchUserInput::username));
-		std::strncpy(input.zone, zone.c_str(), sizeof(SwitchUserInput::zone));
+		std::strncpy(input.username, _username.c_str(), sizeof(SwitchUserInput::username) - 1);
+		std::strncpy(input.zone, zone.c_str(), sizeof(SwitchUserInput::zone) - 1);
```

Comment:
can we get a comment on these minus 1's?

Reply:
Not sure it's worth leaving a comment for these since this is essentially how `strncpy` should be used with null-terminated strings. You always have to leave space for the null byte.

It feels like this should be common knowledge to an experienced C/C++ programmer.

Thoughts?

Reply:
that's perfectly fair. resolve with impunity.

Reply:
I'll see if I can come up with a way to make it obvious. Perhaps just giving -1 a name is the right thing to do. Hmm.

Reply:
Replaced with function call.